### ISSUE TITLE
Allow test runners to accept extra arguments

### ISSUE THREAD

**Issue by abingham** [sixty-north/cosmic-ray#129]
We need to allow users to somehow pass extra arguments to test runners. For example, when testing hypothesis I've seen that its test suite consistently produces 9 errors which are expected and acceptable. But since we fail on first error for py.test, we never actually run any tests for hypothesis.

We need to let users somehow be able to pass arguments like "--maxfail=XXX" to test runners so that tests can be tailored as necessary.

**Comment by abingham** [comment]
This is pretty important, it seems. We need a more sophisticated overall strategy for letting users specify which modules to test, which tests to run, and how to configure those tests. For example, when using CR with segpy we found that we wanted to be able to run just the test_float.py tests, and this required modifying the CR code. That kind of thing should be possible extrinsically.

**Comment by abingham** [comment]
I think we can do this pretty easily. On init the user should be able to pass arbitrary extra arguments that get passed verbatim to the test-runner constructor. These would be stored in the test-runner base class and could be used as needed (including being rejected, I suppose) by the subclass.

So perhaps something like this:
```
cosmic-ray init --baseline=2 --test-runner=pytest sess1 moduleA -- --maxfail=10 -k test_special moduleA/test
```

**Comment by abingham** [comment]
Fixed in #141
